editor/editor_help.cpp

```diff
@@ -43,8 +44,53 @@
 
 #define CONTRIBUTE_URL vformat("%s/contributing/documentation/updating_the_class_reference.html", VERSION_DOCS_URL)
 
+// TODO: this is sometimes used directly as doc->something, other times as EditorHelp::get_doc_data(); Might this be a problem?
 DocTools *EditorHelp::doc = nullptr;
 
+static bool _attempt_doc_load(const String &p_class) {
+	// Docgen always happens in the outer-most class: it also generates docs for inner classes.
+	String outer_class = p_class.get_slice(".", 0);
+	if (!ScriptServer::is_global_class(outer_class)) {
+		return false;
+	}
+
+	Ref<Script> script = ResourceLoader::load(ScriptServer::get_global_class_path(outer_class), outer_class);
+	// Check whether docgen happens in ResourceLoader::load() / whether docs are already cached
+	// If not, call reload() which docgens for GDScript
+	// TODO: could be improved with a docgen-only method, since reload() also compiles GDScripts
+	if (script.is_valid() && (script->are_docs_generated() || (script->reload(true) == OK && script->are_docs_generated()))) {
+		Vector<DocData::ClassDoc> docs = script->get_documentation();
```

Comment:
So upon reflection and digging in a little more, it seems that `ResourceLoader::load()` will load the full script, which _for GDScript_ already calls `reload()` and generates documentation. So I believe we might be able to loosen and simplify these checks!

Reply:
Yes, unless the script is already cached, it should call `reload`, which calls the compiler, which should generate the docs. So at the very least, the new `are_docs_generated` method and the attempt at reloading it here seem unnecessary?

Reply:
Yes, it does! I have removed the changes to `Script` and its descendants, which also makes compiling this PR much faster! :)

For posterity, some more thoughts:

The alternative to `ResourceLoader` is to use the stuff in `gdscript_cache.h`, but that is not script-agnostic. However, other script languages don't have similar loading capabilities AFAIK, and because this "shallow" loading cannot happen in the `Script` class and its derivatives, it'd be hard/impossible to abstract it into that.

In the end I think the best compromise is to take the slight inefficiency of using the `ResourceLoader` abstraction, which unnecessarily compiles gdscripts in order to get their documentation.